spec.html

```diff
@@ -6497,13 +6493,12 @@ <h1>
     <emu-clause id="sec-getfunctionrealm" type="abstract operation">
       <h1>
         GetFunctionRealm (
-          _obj_: unknown,
+          _obj_: a function object,
```

Comment:
should this lack "object" per #2489?

Also, does this cover all callables, including exotic host callable objects and Proxies?

Reply:
> should this lack "object" per #2489?

Well, this PR is against #2483, which is currently using "a function object" for parameters. (And anyhow, it looks like #2489 isn't changing that.)
 
> Also, does this cover all callables, including exotic host callable objects and Proxies?

The spec doesn't actually define "callable", but if your criterion is "IsCallable returns true", then that means "an Object with a [[Call]] internal method", which is synonymous with "function object", so yes, "a function object" covers all callables.

(I've been thinking that maybe the spec should drop the term "callable"; its use suggests that it's somehow distinct from "function", but I doubt there's a reasonable definition of it that would be distinct.)

Reply:
A Proxy to a function is a function object? That doesn’t seem right, since it lacks the internal slots of a function (Function toString will throw)

Reply:
> since it lacks the internal slots of a function

No, it [has them](https://tc39.es/ecma262/multipage/#sec-proxy-object-internal-methods-and-internal-slots-call-thisargument-argumentslist).

> Function toString will throw

No it won't; try it yourself:

```js
Function.prototype.toString.call(new Proxy(function(){}, {}))
```

Reply:
wat, are *both* function and array both the special ones that can’t be distinguished from proxies to them? I thought it was just array.

Reply:
Yup, though in slightly different ways: for arrays `isArray` explicitly pierces proxies, whereas for functions the relevant internal slots are added to the proxy itself.